services/repository/mongodb/service.go

```diff
@@ -0,0 +1,204 @@
+// Package mongodb implements MongoDB repository services
+package mongodb
+
+import (
+	"context"
+
+	"github.com/decentralized-cloud/tenant/models"
+	"github.com/decentralized-cloud/tenant/services/configuration"
+	"github.com/decentralized-cloud/tenant/services/repository"
+	commonErrors "github.com/micro-business/go-core/system/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type mongodbRepositoryService struct {
+	configurationService configuration.ConfigurationContract
+	collectionName       string
+}
+
+// NewMongodbRepositoryService creates new instance of the RepositoryService, setting up all dependencies and returns the instance
+// Returns the new service or error if something goes wrong
+func NewMongodbRepositoryService(
+	configurationService configuration.ConfigurationContract) (repository.RepositoryContract, error) {
+
+	if configurationService == nil {
+		return nil, commonErrors.NewArgumentNilError("configurationService", "configurationService is required")
+	}
+
+	return &mongodbRepositoryService{
+		configurationService: configurationService,
```

Comment:
just read the connection string once and only keep the reference to that as part of the service context, no need to keep the reference to the configuration service here

also, read the name of the database here, no need to keep reading it using configuration service